### PR TITLE
remove request listener to prevent untimely session invalidation

### PR DESCRIPTION
**Pull request by DaanHoogland**
### Description

This PR removes the request handlin from the ApiSessionListener. This causes sessions to be invalidated. API calling from cmk would fail and succeed on and off due to this.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

#### Bug Severity

- [x] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### How Has This Been Tested?

manually tested by listing accounts with cloudmonkey.

### CODE REVIEW

**Comment by nvazquez** [comment]
@blueorangutan package

**Comment by nvazquez** [comment]
@blueorangutan test

**Comment by DaanHoogland** [comment]
@weizhouapache @nvazquez I am ok with this as a temporary fix, but it is a functional regression as sessions are no longer properly administrated. We (I) will have to think about a more premanent fix. I actually think the real bug is in the way the ApiServlet handles sessions and not this listener, but they seem to be incompatible :(

**Review by sureshanaparti — APPROVED** [review]
code LGTM

**Review by rohityadavcloud — APPROVED** [review]

File: server/src/main/java/com/cloud/api/ApiSessionListener.java [hunk 1]
```diff
@@ -55,29 +52,12 @@ public void sessionCreated(HttpSessionEvent event) {
...
-    public void requestDestroyed(ServletRequestEvent event) {
-        LOGGER.debug("request destroyed");
-    }
-
-    @Override
-    public void requestInitialized(ServletRequestEvent event) {
-        LOGGER.debug("request initialized");
-        HttpServletRequest request = (HttpServletRequest) event.getServletRequest();
-        HttpSession session = request.getSession();
-        if (session.isNew()) {
```
Comment (rohityadavcloud) [inline server/src/main/java/com/cloud/api/ApiSessionListener.java:76]: @DaanHoogland maybe one way is to use a ConcurrentHashMap to store total unique sessions?

Comment (DaanHoogland) [reply server/src/main/java/com/cloud/api/ApiSessionListener.java:76]: not sure what you mean. It is already used like that , is it?

**Comment by DaanHoogland** [comment]
@blueorangutan package

File: server/src/main/java/com/cloud/api/ApiServlet.java [hunk 2]
```diff
@@ -186,22 +190,28 @@ void processRequestInContext(final HttpServletRequest req, final HttpServletResp
...
             }
 
             final Object[] commandObj = params.get(ApiConstants.COMMAND);
-            if (commandObj != null) {
-                final String command = (String) commandObj[0];
+            final String command = commandObj == null ? null : (String) commandObj[0];
+            final Object[] userObj = params.get(ApiConstants.USERNAME);
+            String username = userObj == null ? null : (String)userObj[0];
+            if (s_logger.isTraceEnabled()) {
+                s_logger.trace(String.format("command %s processing for user \"%s\"", command, username));
```
Comment (DaanHoogland) [inline server/src/main/java/com/cloud/api/ApiServlet.java:209]:
```suggestion
                s_logger.trace(String.format("command %s processing for user \"%s\"", command, username.replaceAll("[\n\r\t]", "_")));
```

File: server/src/main/java/com/cloud/api/ApiServlet.java [hunk 3]
```diff
@@ -229,6 +236,10 @@ void processRequestInContext(final HttpServletRequest req, final HttpServletResp
                     }
 
                     try {
+                        if (s_logger.isTraceEnabled()) {
+                            s_logger.trace(String.format("apiAuthenticator.authenticate(%s, %s, %s, %s, %s, %s, %s,%s)",
+                                    command, params, session, remoteAddress, responseType, auditTrailSb, req, resp));
```
Comment (DaanHoogland) [inline server/src/main/java/com/cloud/api/ApiServlet.java:241]:
```suggestion
                            s_logger.trace(String.format("apiAuthenticator.authenticate(%s, params[%d], %s, %s, %s, %s, %s,%s)",
                                    command, params.size(), session, remoteAddress, responseType, auditTrailSb, req, resp));
```

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
tested @nvazquez @weizhouapache ,
both repeated listing of accounts and (with latency) session count works.

**Review by nvazquez — APPROVED** [review]

File: server/src/main/java/com/cloud/api/ApiServlet.java [hunk 4]
```diff
@@ -186,22 +193,30 @@ void processRequestInContext(final HttpServletRequest req, final HttpServletResp
         }
 
         try {
-
-            if (HttpUtils.RESPONSE_TYPE_JSON.equalsIgnoreCase(responseType)) {
-                resp.setContentType(ApiServer.JSONcontentType.value());
-            } else if (HttpUtils.RESPONSE_TYPE_XML.equalsIgnoreCase(responseType)){
-                resp.setContentType(HttpUtils.XML_CONTENT_TYPE);
-            }
+            resp.setContentType(HttpUtils.XML_CONTENT_TYPE);
```
Comment (rohityadavcloud) [inline server/src/main/java/com/cloud/api/ApiServlet.java:196]: @DaanHoogland should we force xml?

**Comment by rohityadavcloud** [comment]
Test cases suggested:
- Normal login as different account/roles/types
- Login/auth and run API using api-key and secret-keys
- SAML based SSO login and logout
- pre-shared key based running APIs

File: server/src/main/java/com/cloud/api/ApiServlet.java (hunk 4 above)
Comment (DaanHoogland) [reply server/src/main/java/com/cloud/api/ApiServlet.java:196]: It wouldn´t matter in the existing code as line 179 (former version 168) https://github.com/apache/cloudstack/pull/6393/files#diff-cc92beff292fdad7835437d622159db6e1ae90d0ebac7fb4ae73c813faa8896eR179 hard-codes the `reponseType` to `HttpUtils.RESPONSE_TYPE_XML`.  The second branch would always be executed.

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Comment by rohityadavcloud** [comment]
@blueorangutan test

**Comment by nvazquez** [comment]
Hi @kohrar can you please verify the login/logout with this PR?

**Pull request merged**